src/config.rs

```diff
@@ -50,6 +50,9 @@ pub struct Hook {
 	/// The commands to execute when the hook is triggered.
 	pub commands: Vec<Command>,
 
+	/// The environment variables to set when the hook is triggered.
+	pub environment: Option<Vec<String>>,
```

Comment:
```suggestion
    #[serde(default)]
	pub environment: IndexMap<String, String>,
```

Let's use an `IndexMap` (you'll have to add the `indexmap` dependency). Then we can just use the YAML key as the environment variable name, and `IndexMap` still preserves the order (although that's probably not super important, it's still nice).

Reply:
I placed the #[serde(default)] above environment as you specified.
```
error[E0277]: the trait bound `IndexMap<std::string::String, std::string::String>: Deserialize<'_>` is not satisfied
    --> src\config.rs:56:19
     |
56   |     pub environment: IndexMap<String, String>,
     |                      ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `IndexMap<std::string::String, std::string::String>`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               bool
               char
               isize
               i8
               i16
               i32
               i64
               i128
             and 147 others
note: required by a bound in `next_element`
    --> C:\Users\dhorner\.cargo\registry\src\index.crates.io-6f17d22bba15001f\serde-1.0.188\src\de\mod.rs:1724:12
     |
1722 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1723 |     where
1724 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`
```

Reply:
Ah, yes, you need to enable the `serde` feature of `indexmap`:
```sh
cargo add indexmap --features serde
```